engine/src/iutil.cpp

```diff
@@ -977,14 +977,14 @@ bool MCImageRotateRotate(MCImageBitmap *p_src, real64_t p_angle, uint32_t p_back
 
 			if (quality == INTERPOLATION_BOX)
 			{
-				if (t_ix < 0 || t_ix >= p_src->width || t_iy < 0 || t_iy >= p_src->height)
+				if (t_ix < 0 || (uint32_t)t_ix >= p_src->width || t_iy < 0 || (uint32_t)t_iy >= p_src->height)
 					t_pixel = p_backing_color;
 				else
 					t_pixel = MCImageBitmapGetPixel(p_src, t_ix, t_iy);
 			}
 			else
 			{
-				if (t_ix < -1 || t_ix >= p_src->width || t_iy < -1 || t_iy >= p_src->height)
+				if (t_ix < -1 || (uint32_t)t_ix >= p_src->width || t_iy < -1 || (uint32_t)t_iy >= p_src->height)
```

Comment:
Doesn't help that it's been like that since the file was initially committed to git!